### PR TITLE
OBPIH-3860 Partially invoiced status in order item and order summary …

### PR DESCRIPTION
**Pull request by awalkowiak**
…views

### CODE REVIEW

File: grails-app/migrations/views/order-summary-view.sql [hunk 1]
```diff
@@ -2,50 +2,46 @@ CREATE OR REPLACE VIEW order_item_status AS
...
 CREATE OR REPLACE VIEW order_receipt_status AS
     SELECT
         order_id,
         order_item_id,
         order_number,
-        quantity_ordered,
-        quantity_received,
-        quantity_canceled
+        SUM(quantity_received) AS quantity_received,
+        SUM(quantity_canceled) AS quantity_canceled
```
Comment (jmiranda) [inline grails-app/migrations/views/order-summary-view.sql:35]: What's the reason behind removing quantity ordered. Are we just getting that from a different view?

File: grails-app/migrations/views/order-summary-view.sql [hunk 2]
```diff
@@ -55,74 +51,65 @@ CREATE OR REPLACE VIEW order_receipt_status AS
...
         order_number,
         order_item_id,
-        quantity_ordered,
-        shipment_item_quantity_invoiced,
-        order_status,
-        (quantity_ordered > 0)                                AS shipment_payment_ordered,
-        (shipment_item_quantity_invoiced >= quantity_ordered) AS shipment_item_invoiced,
-        invoice_submitted
+        SUM(quantity_ordered)  				AS quantity_ordered,
+        IFNULL(SUM(quantity_invoiced), 0) 	AS quantity_invoiced
```
Comment (jmiranda) [inline grails-app/migrations/views/order-summary-view.sql:62]: I wonder if it would be a good iea to use the IFNULL convention for all of these SUMs.

**Review by jmiranda — COMMENTED** [review]
All of these views are fine for now because we don't have many orders or invoices. But eventually we might start to see performance degradation and it'll happen quickly. So we should have a plan of attack for creating materialized views or some other way to store / retrieve this data that doesn't require building a temporary table for every query.

File: grails-app/migrations/views/order-summary-view.sql (hunk 1 above)
Comment (awalkowiak) [reply grails-app/migrations/views/order-summary-view.sql:35]: I think I wanted to base the quantity ordered on the order item summary, and not have the same quantities in each of these views and keep each "minor" view tied to specific values:
- **order_item_status** for quantity_ordered and quantity_shipped,
- **order_receipt_status** for quantity_received and quantity_canceled,
- **order_item_payment_status** for quantity_invoiced for order items,
- **order_adjustment_payment_status** for quantity_ordered (which is used for adjustment items count), quantity_invoiced (which is used for adjustment items invoiced count)

File: grails-app/migrations/views/order-summary-view.sql (hunk 2 above)
Comment (awalkowiak) [reply grails-app/migrations/views/order-summary-view.sql:62]: I added it in some other PR, because it turned out it was a good idea

**Pull request merged**